base/show.jl

```diff
@@ -3128,6 +3128,24 @@ function array_summary(io::IO, a, inds)
     print(io, " with indices ", inds2string(inds))
 end
 
+"""
+    summary(io::IO, ref::GenericMemoryRef)
+
+Print a brief description of a `GenericMemoryRef`, showing the remaining
+accessible length from the current offset. This improves error messages
+like `BoundsError` to reflect the effective size, similar to `SubString`.
+"""
+function Base.summary(io::IO, ref::GenericMemoryRef{kind,T,AS}) where {kind,T,AS}
+    mem = parent(ref)
+    idx = Core.memoryrefoffset(ref)            # 1-based offset within parent memory
+    rem = length(mem) - (idx - 1)              # remaining accessible elements
```

Comment:
I prefer the variable naming in https://github.com/JuliaLang/julia/pull/57278/files